docs/design/commonevent-trigger.md

```diff
@@ -0,0 +1,104 @@
+# Add trigger template for CommonEvent
+
+## Abstract 
+
+We currently have two ways to trigger the diagnosis process, one is by monitoring kubernetes events and the other is by receiving Prometheus alert templates events. The PagerDuty Common Event Format (PD-CEF) is a standardized alert format that allows you to view alert and incident data in a cleaner, more normalized way. They express common event concepts in a normalized, readable way.
+
+The goal of this Task is to implement the trigger that supports Common Event, the diagnosis is automatically generated based on CommonEvent Template to trigger the diagnosis process.
+
+## Design
+
+The table below outlines the name and type of each PD-CEF field. See more details at <https://support.pagerduty.com/docs/pd-cef>.
+
+|        Name        |                 Type                 |
+| :----------------: | :----------------------------------: |
+|    **Summary**     |                String                |
+|     **Source**     |                String                |
+|    **Severity**    | enum{Info, Warning, Error, Critical} |
+|   **Timestamp**    |              Timestamp               |
+|     **Class**      |                String                |
+|   **Component**    |                String                |
+|     **Group**      |                String                |
+| **Custom Details** |                Object                |
+
+Considering all the above fields, the following CommonEventFormat struct is designed:
+
+```go
+type CommonEventFormat struct {
```

Comment:
It could be referred to https://github.com/kubediag/kubediag/blob/master/api/v1/commonevent_types.go#L24 instead of redesign.